src/agent/src/mount.rs

```diff
@@ -133,6 +133,50 @@ lazy_static! {
     ];
 }
 
+#[derive(Debug)]
+pub struct RefCountedObject<T> {
+    pub name: String,
+    pub ref_count: u32,
+    pub ref_type: T,
+}
+#[derive(Debug, Copy, Clone)]
+// Define an enum to represent the MountPoint and VerityDevice variants
+pub enum ObjectKind {
+    MountPoint,
+    VerityDevice,
+}
+// Implement the RefCounted trait for the generic struct
+impl RefCountedObject<ObjectKind> {
+    #[inline]
+    pub fn inc_ref(&mut self) {
+        self.ref_count += 1;
+    }
+    pub fn dec_ref(&mut self) -> Result<()> {
```

Comment:
I think we should shield the specific implementation and let the storage implement its own method of increment and decrement the reference count, so that different storages can manage their own processing when the reference count is 0. Mountpoint can remove mounts, but verity devices can close by themselves.